src/flamegpu/model/DependencyGraph.cpp

```diff
@@ -265,9 +265,13 @@ void DependencyGraph::checkForUnattachedFunctions() {
         }
     };
 
+    for (auto& root : roots) {
+        captureFunctions(root);
+    }
+
     // Compare sets
     if (modelFunctions != graphFunctions) {
-        std::cout << "WARNING: Not all agent functions are used in the dependency graph - have you forgotten to add one?";
+        std::cout << "WARNING: Not all agent functions are used in the dependency graph - have you forgotten to add one?" << std::flush;
```

Comment:
Should this be `cerr`? (Honestly not sure what team preference would be).

Reply:
Warnigns / erros shoudl go to stderr IMO. We've tended to use `fprintf` rather than `std::cout/std::cerr` although that's mostly just habit (although there's generally a lot of dislike in c++ for output streams for some reason?)

Reply:
Yeah, i don't think use of std::cout vs printf matter, I was mostly concerned that warnings probably don't want to stderr rather than stdout.

In general, console IO has been minimised in the codebase, preferring to throw exceptions. But this warning seems reasonable, otherwise users would need to specify they don't want to use an agent function, or comment it out which seems a bit arkward.

Reply:
I have updated to use cerr in light of this.